### PR TITLE
Porting engine to ARM64 architecture

### PR DESCRIPTION
**Pull request by Zigatun**

### CODE REVIEW

File: src/xrCore/Math/MathUtil.cpp [hunk 1]
```diff
@@ -48,9 +50,16 @@ void Initialize()
...
+#if defined(XR_ARM64)
+	PLCCalc = PLCCalc_CPP;
+#else
+	PLCCalc = PLCCalc_SSE;
+#endif
     //PLCCalc = PLCCalc_CPP;
 #endif
+
+
+
```
Comment (Xottab-DUTY) [inline src/xrCore/Math/MathUtil.cpp]: Please, remove that empty added space.

Comment (Zigatun) [reply src/xrCore/Math/MathUtil.cpp]: I fixed this.

File: src/xrCore/Math/MathUtil.cpp [hunk 2]
```diff
@@ -48,9 +50,14 @@ void Initialize()
     Skin2W = Skin2W_CPP;
     Skin3W = Skin3W_CPP;
     Skin4W = Skin4W_CPP;
-    PLCCalc = PLCCalc_SSE;
+#if defined(XR_ARM64)
+	PLCCalc = PLCCalc_CPP;
```
Comment (q4a) [inline src/xrCore/Math/MathUtil.cpp:54]: Replace tabs with 4 space, please. And in else section.
```suggestion
    PLCCalc = PLCCalc_CPP;

File: src/xrCommon/inlining_macros.h [hunk 3]
```diff
@@ -4,7 +4,7 @@
 #define NO_INLINE __attribute__((noinline))
 #define FORCE_INLINE __attribute__((always_inline)) inline
 #define ALIGN(a) __attribute__((aligned(a)))
-#define DEBUG_BREAK asm("int $3")
+#define DEBUG_BREAK __builtin_trap()
```
Comment (q4a) [inline src/xrCommon/inlining_macros.h:7]: Nice change, thanks!

File: Externals/sse2neon/sse2neon.h [hunk 4]
```diff
@@ -0,0 +1,2793 @@
+#ifndef SSE2NEON_H
```
Comment (q4a) [inline Externals/sse2neon/sse2neon.h:1]: Did you take it from https://github.com/DLTcollab/sse2neon ?

File: src/xrCore/Math/MathUtil.cpp [hunk 5]
```diff
@@ -48,9 +50,14 @@ void Initialize()
...
+#else
+	PLCCalc = PLCCalc_SSE;
```
Comment (q4a) [inline src/xrCore/Math/MathUtil.cpp:56]:
```suggestion
    PLCCalc = PLCCalc_SSE;

File: src/xrCore/_math.cpp [hunk 6]
```diff
@@ -208,7 +217,13 @@ XRCORE_API u64 QPC() noexcept
 
 XRCORE_API u64 GetCLK()
 {
+#if defined(XR_ARM64)
+  int64_t virtual_timer_value;
+  asm volatile("mrs %0, cntvct_el0" : "=r"(virtual_timer_value));
+  return virtual_timer_value;
```
Comment (q4a) [inline src/xrCore/_math.cpp:223]: Use 4 space, please.
```suggestion
    int64_t virtual_timer_value;
    asm volatile("mrs %0, cntvct_el0" : "=r"(virtual_timer_value));
    return virtual_timer_value;

**Review by q4a — CHANGES_REQUESTED** [review]
Please, fix few spaces.

**Pull request merged**